src/bridge/ERC20Inbox.sol

```diff
@@ -139,4 +164,21 @@ contract ERC20Inbox is AbsInbox, IERC20Inbox {
                 tokenAmount
             );
     }
+
+    /// @inheritdoc AbsInbox
+    function _fromNativeTo18Decimals(uint256 value) internal view override returns (uint256) {
+        // In order to keep compatibility of child chain's native currency with external 3rd party tooling we
+        // expect 18 decimals to be always used for native currency. If native token uses different number of
+        // decimals then here it will be normalized to 18. Keep in mind, when withdrawing from child chain back
+        // to parent chain then the amount has to match native token's granularity, otherwise it will be rounded
+        // down.
+
+        // Also make sure that inflated amount does not overflow uint256
+        if (nativeTokenDecimals < 18) {
+            if (value > MAX_BRIDGEABLE_AMOUNT) {
+                revert AmountTooLarge(value);
```

Comment:
would be nice to also enforce this in nitro so that withdrawal that exceed this size (unlikely) would not be stuck